_posts/2019-04-06-trees-and-rectangles-reconciling-sql-and-recursive-programming-languages.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: post
+title: "trees and rectangles: reconciling SQL and recursive programming languages"
+description: ""
+category:
+tags: [haskell, sql]
+---
+
+There's a contradiction at the heart of web development.
+
+Most structures from HTML through to JSON calls are fundamentally
+hierarchical. Relationships are expressed primarily through
+embedding, and this means we mostly deal with trees.
```

Comment:
This sentence might need a bit more development... I think you're saying that all programming is mapping trees of one shape into trees of another?  But maybe not...

Reply:
yeah, i guess i've implicitly scoped it to web dev. i should call that out.